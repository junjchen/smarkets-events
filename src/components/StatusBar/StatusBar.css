.status-bar {
    font-size: 0.8em;
    padding: 0.5em;
    color: rgb(255, 255, 255);
    background-color: rgb(67, 160, 71);
}

.status-bar p {
    margin: 0.5em 0;
}

.status-bar--error {
    background-color: rgb(244, 81, 30);
}

.status-bar--fetching {
    background-color: rgb(117, 117, 117);
}