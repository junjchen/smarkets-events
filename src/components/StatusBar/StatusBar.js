import React, {Component} from 'react'
import classNames from 'classnames'
import moment from 'moment'
import './StatusBar.css'

class StatusBar extends Component {
    render() {
        const {state, reason, updated} = this.props
        const fetching = state === 'loading'
        const error = state === 'error'
        const ok = state === 'ok'
        return (
            <div
                className={classNames('status-bar', {
                'status-bar--error': error,
                'status-bar--fetching': fetching
            })}>
                {ok && <p>Ready</p>}
                {fetching && <p>Fetching...</p>}
                {error && <p>Oops!</p>}
                {reason && <p>{reason.name}</p>}
                {updated && <p>Last updated: {moment(updated).format('MMMM Do YYYY, h:mm:ss a')}</p>}
            </div>
        )
    }
}

export default StatusBar