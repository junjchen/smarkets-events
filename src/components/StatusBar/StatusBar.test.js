import React from 'react'
import {shallow} from 'enzyme'
import moment from 'moment'
import StatusBar from './StatusBar'

const createProps = (state, reason) => ({state, reason, updated: new Date('1990/10/10')})

describe('StatusBar', () => {
    it('renders same as snapshot', () => {
        const props = createProps()
        const wrapper = shallow(<StatusBar {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders fetching status', () => {
        const props = createProps('loading')
        const wrapper = shallow(<StatusBar {...props}/>)
        expect(wrapper.hasClass('status-bar--fetching')).toBe(true)
        expect(wrapper.contains(
            <p>Fetching...</p>
        )).toBe(true)
    })
    it('renders ok status', () => {
        const props = createProps('ok')
        const wrapper = shallow(<StatusBar {...props}/>)
        expect(wrapper.contains(
            <p>Ready</p>
        )).toBe(true)
    })
    it('renders error status', () => {
        const props = createProps('error', {name: 'reason'})
        const wrapper = shallow(<StatusBar {...props}/>)
        expect(wrapper.hasClass('status-bar--error')).toBe(true)
        expect(wrapper.contains(
            <p>Oops!</p>
        )).toBe(true)
        expect(wrapper.contains(
            <p>reason</p>
        )).toBe(true)
    })
})
