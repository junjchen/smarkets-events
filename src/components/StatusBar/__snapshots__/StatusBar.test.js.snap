// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders same as snapshot 1`] = `
<div
  className="status-bar"
>
  <p>
    Last updated: 
    October 10th 1990, 12:00:00 am
  </p>
</div>
`;
